src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,8 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            internal FixedString.Size260 cFileName;
+            internal FixedString.Size14 cAlternateFileName;
```

Comment:
> A property off of a field like this hides it

That is a factor of naming and using properties vs. methods not something inherently wrong with wrapping in a struct vs. not wrapping in a struct. I chose not to make it ToString()/FromString() as a starting point given what I expected usage to be like, but that is easy enough to change. You can do what you can to mitigate over-allocating but you'll never eliminate the problem- you *can* eliminate mismatching the buffer size with my solution.

I still don't get how it is such a big deal to encapsulate "tricky" behavior. I'd call this a lot less tricky and more transparent than

``` C#
[MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
internal string cAlternateFileName;
```

This native struct has 2 fixed char arrays, some have more. It is super easy to make a mistake when copying accessors and not get the right size and not see the problem, potentially for a long time. Wrapping into a `Span<char>` isn't too bad, and I'm willing to do that- but it doesn't fix the multiple size problem. C# can fix that problem (long term) in this case, which is what pushes me over the edge here.

FWIW, my ultimate solution here would be something like:

```
internal fixed Span<char>[260] cFileName;
```